src/smif/controller/scheduler.py

```diff
@@ -0,0 +1,88 @@
+""" Implements methods for loading, building and executing model runs
+and a Scheduler that runs instances of smif as a subprocess
+"""
+from subprocess import PIPE, Popen
+
+
+class Scheduler(object):
+
+    def __init__(self):
+        self._process = {}
+        self._output = {}
+        self._err = {}
+
+    def add(self, model_run_name, args):
+        """Add a model_run to the Modelrun scheduler.
+
+        Parameters
+        ----------
+        model_run_name: str
+            Name of the modelrun
+        args: dict
+            Arguments for the command-line interface
+
+        Exception
+        ---------
+        Exception
+            When the modelrun was already started
```

Comment:
Should this also currently fail here if ANY modelrun has already started?

Reply:
No it is possible to run models concurrently. As discussed we will leave this like this now.